src/main.rs

```diff
@@ -41,16 +48,29 @@ fn handle_search(settings: &Settings, history: &History) {
     let result = Interface::new(settings, history).display();
     if let Some(cmd) = result.selection {
         if let Some(path) = &settings.output_selection {
-            // Output selection to a file, with the first line indicating if the user chose to run the selection or not.
+            // Output selection results to a file.
             let mut out: String = String::new();
 
+            // First we say the desired mode, depending on the key pressed by the user - simply
+            // displaying the selected command, or running it.
             if result.run {
-                out.push_str("run\n");
+                out.push_str("mode run\n");
             } else {
-                out.push_str("display\n");
+                out.push_str("mode display\n");
             }
 
+            // Next, the desired commandline selected by the user.
+            out.push_str("commandline ");
```

Comment:
Do multi-line commands work?

Reply:
Not exactly.  I don't think they work currently for zsh either, so I thought it'd be better addressed in a separate change, what do you think?

fish does a bit better than zsh.  The full command gets stored in history.  It shows up in the interface when you have a matching search; it seems to indent later lines more than necessary, but it's all there.  When you select the command, though, you only get the first line.

zsh stores multi-line commands on multiple lines in the history file (latter lines with no extended timestamp) and we currently assume each line is one command, so you'd get multi-line commands broken (figuratively and literally) into multiple.  We could be a bit cheeky and assume that if we're using extended timestamps and the next line doesn't include one, it's part of the previous command... the alternative seems to be actually understanding what zsh considers a complete command, which might get ugly.

(Not sure about bash)

Reply:
Yea, I think you're right. Multiline support is pretty broken across the board.